Design/spectrum_number_and_workspace_index_abstraction.md

```diff
@@ -0,0 +1,269 @@
+# Indexing Abstraction Layer
```

Comment:
Ideally, I think workspace/spectra index needs to be banished from the user interface (both graphical/non-graphical) so that it's purely a developer thing and then I think we have more scope for renaming the concept.